sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InjectRuntimeFilter.scala

```diff
@@ -99,16 +99,17 @@ object InjectRuntimeFilter extends Rule[LogicalPlan] with PredicateHelper with J
       filterCreationSideExp: Expression,
       filterCreationSidePlan: LogicalPlan): LogicalPlan = {
     require(filterApplicationSideExp.dataType == filterCreationSideExp.dataType)
-    val actualFilterKeyExpr = mayWrapWithHash(filterCreationSideExp)
-    val alias = Alias(actualFilterKeyExpr, actualFilterKeyExpr.toString)()
+    val alias = Alias(filterCreationSideExp, filterCreationSideExp.toString)()
     val aggregate = ColumnPruning(Aggregate(Seq(alias), Seq(alias), filterCreationSidePlan))
```

Comment:
shall we simply add the hash expression for the second `Seq(alias)`?

Reply:
This will increase the calculation.

Reply:
I have ran the TPC-DS with you suggestion, regression appears.

Reply:
I don't get it. How is `Project(expr, Agg(...))` different from `Agg(grouping, expr)`? `Agg` has a project list and can do the job of `Project` as well.

Reply:
I'm sorry. I forgot `spark.sql.optimizer.runtime.bloomFilter.enabled=false`.

Reply:
I will test again.